### DISCUSSION TITLE
split: is there way to set output folder?

### DISCUSSION THREAD

**Discussion by aborruso** [johnkerl/miller#1402] [Q&A] [answered]
Hi,
I know, I can run this from the output folder `mlr --csv --from /tmp/sample/input.csv split -n 1000000`

Is there way to set an output folder that is not the current one?

Thank you

**Comment by johnkerl** [comment, marked as answer]
@aborruso I think you want `--prefix` (and the online help can be clearer about this fact):

```
$ mlr --d2p split -n 2 --prefix /tmp/abor data/small
```

```
$ ls -l /tmp/abor*
-rw-r--r--  1 johnkerl  wheel  127 Sep 26 08:14 /tmp/abor_1.pprint
-rw-r--r--  1 johnkerl  wheel  132 Sep 26 08:14 /tmp/abor_2.pprint
-rw-r--r--  1 johnkerl  wheel  128 Sep 26 08:14 /tmp/abor_3.pprint
-rw-r--r--  1 johnkerl  wheel  126 Sep 26 08:14 /tmp/abor_4.pprint
-rw-r--r--  1 johnkerl  wheel  133 Sep 26 08:14 /tmp/abor_5.pprint
```

```
$ for x in /tmp/abor*; do echo; echo $x; cat $x; done

/tmp/abor_1.pprint
a   b   i x                  y
pan pan 1 0.3467901443380824 0.7268028627434533
eks pan 2 0.7586799647899636 0.5221511083334797

/tmp/abor_2.pprint
a   b   i x                   y
wye wye 3 0.20460330576630303 0.33831852551664776
eks wye 4 0.38139939387114097 0.13418874328430463

/tmp/abor_3.pprint
a   b   i x                  y
wye pan 5 0.5732889198020006 0.8636244699032729
zee pan 6 0.5271261600918548 0.49322128674835697

/tmp/abor_4.pprint
a   b   i x                  y
eks zee 7 0.6117840605678454 0.1878849191181694
zee wye 8 0.5985540091064224 0.976181385699006

/tmp/abor_5.pprint
a   b   i  x                   y
hat wye 9  0.03144187646093577 0.7495507603507059
pan wye 10 0.5026260055412137  0.9526183602969864
```

**Reply by aborruso** [reply]
> @aborruso I think you want `--prefix` (and the online help can be clearer about this fact):

Yes, because in the text we have "Specify filename prefix". I will add something like "Specify filename prefix (you can add output directory)"

Thank you very much

**Comment by tgirod** [comment]
I'm trying something similar. My use case is storing the splitted files in a subfolder.

I can write `mlr split --prefix subfolder/` but all my files will be prefixed with an underscore I do not want. I see no way around it. Maybe a `--folder` option to split would be useful ?
